### ISSUE TITLE
Not sure, think calib3d's calibration.cpp broke between 4.1.0 and 4.1.1

### ISSUE THREAD

**Issue by erichfrazer** [opencv/opencv#15910]
- OpenCV => 4.1.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2019

I've been looking at the code for stereoRectify for a long time. It's returning a lot of nonsense values for some of the calibrated cameras we have. I noticed the code changed from 4.1.0 to 4.1.1, in stereoRectify and it looks like the acos(x) stuff in there is silly and/or doesn't work. If whoever put that code in there really thinks its valid, how about some comments, because what is in there is really weird.

![calibration](https://user-images.githubusercontent.com/15388012/68809595-1b4dea80-0621-11ea-9df9-494284c12666.JPG)

**Comment by alalek** [comment]
> looks like the acos(x) stuff in there is silly and/or doesn't work

Provide complete minimal reproducer including input data and expected results.

**Comment by alalek** [comment]
relates #14718
There is link for better investigation: https://github.com/opencv/opencv/blame/4.1.1/modules/calib3d/src/calibration.cpp#L2533-L2539

**Comment by erichfrazer** [comment]
Actually, I don't have the time to do that right now. I'm surprised there aren't any issues against this code. What I'm saying is whoever wrote that code, should put in some comments, because to me, amateur math whiz that I am, it looks wrong. I just haven't proven so yet.

**Comment by erichfrazer** [comment]
I don’t understand what you just pointed me to.

I guess that person checked in two bits of code into calibrate.cpp, one which was better, one which was wrong.  And yet 4.1.1 has the code in it that looks like it is the unstable code. The code I pointed out before, with the acos stuff in it. I’ve been staring at that code for 24 hours, would love to know why it should have worked. From what I know about math and vectors, it shouldn’t have ever worked consistently.

**Comment by erichfrazer** [comment]
Yes, I realize now why it becomes unstable: for small angle deltas between the unit vector “uu” and the L2R translation vector “t”, the cross product becomes unstable and the answer “blows up”.

The prior version that used two cross products, was more stable.

**Comment by alalek** [comment]
In this case it make sense to add separate branch for small angles:
```
if (<small angle>)
{
    ... more stable 2 cross products approach ...
}
else
{
    ... current code ...
}
```

It would be nice to add some simple regression test for that (see [this](https://github.com/opencv/opencv/pull/14718/commits/62610f1d4c1485fb971fc3c008fdc541b9caa175)).

**Comment by erichfrazer** [comment]
I’d just use the old code, it was technically correct anyhow, just very poorly documented.

I can push a better commented, better variable named version, if anybody allows me to.
